### PR TITLE
Removed support for compliance and a1 server from InSpec compliance

### LINKED ISSUE #4508: Remove support for Compliance and A1 from `inspec compliance`
# Describe the problem

NOTE: this work would begin in Q1 2020 at the earliest.

As Compliance Server has been sunsetted for a while, and Automate 1 (the predecessor to the current product, Automate) [will be sunsetted at the end of 2019](https://blog.chef.io/2019/06/26/end-of-life-announcement-for-chef-automate-v1-x/), once both products are sunsetted, remove code to conditionalize connections to the older two products.

Most of the plugin deals with 3-pronged conditionals; it's quite heavy.

### PR DESCRIPTION
**Pull request by Nik08**
Signed-off-by: Nikita Mathur <nikita.mathur@chef.io>

Removed support for compliance and a1 server from InSpec compliance

## Description

1. Removed support for compliance and a1 server from InSpec compliance by removing the conditional checks for a1 and compliance servers
2. Changes in test cases for InSpec Compliance because of the above change.
## Related Issue

Closes #4508 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by tas50** [comment]
I was just going to ask about this for some of the dist workmark stuff. Glad to see the cleanup

File: lib/plugins/inspec-compliance/lib/inspec-compliance/api.rb [hunk 1]
```diff
@@ -24,19 +24,7 @@ class API
...
-          url = "#{config["server"]}/user/compliance"
-        # Chef Automate2
-        elsif is_automate2_server?(config)
-          url = "#{config["server"]}/compliance/profiles/search"
-        # Chef Automate
-        elsif is_automate_server?(config)
-          url = "#{config["server"]}/profiles/#{owner}"
-        else
-          raise ServerConfigurationMissing
-        end
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-compliance/lib/inspec-compliance/api.rb]: This isn't quite the same - if the server config is missing, that error won't be caught. I think we still need to check for `is_automate2_server?`.

Comment (Nik08) [reply lib/plugins/inspec-compliance/lib/inspec-compliance/api.rb]: Yes! Nice catch. Will preserve this check.

**Comment by Vasu1105** [comment]
LGTM @Nik08.

**Review by Vasu1105 — APPROVED** [review]

**Review by clintoncwolfe — APPROVED** [review]
I tested this against my A2 vm and it worked fine - thanks!

**Pull request merged**